pkg/controller/publishingstrategy/publishingstrategy_controller.go

```diff
@@ -214,297 +272,124 @@ func (r *ReconcilePublishingStrategy) Reconcile(request reconcile.Request) (reco
-func isOnCluster(publishingStrategyIngress *cloudingressv1alpha1.ApplicationIngress, ingressController operatorv1.IngressController) bool {
-	if publishingStrategyIngress.DNSName != ingressController.Spec.Domain {
-		log.Info("ApplicationIngress.DNSName mismatch",
-			"ApplicationIngress", publishingStrategyIngress,
-			"IngressController", ingressController)
-		return false
-	}
-	if publishingStrategyIngress.Certificate.Name != ingressController.Spec.DefaultCertificate.Name {
-		log.Info("ApplicationIngress.Certificate.Name mismatch",
-			"ApplicationIngress", publishingStrategyIngress,
-			"IngressController", ingressController)
-		return false
-	}
-	listening := string(publishingStrategyIngress.Listening)
-	capListening := strings.Title(strings.ToLower(listening))
-	if ingressController.Status.EndpointPublishingStrategy.LoadBalancer == nil && capListening == "Internal" {
-		log.Info("ApplicationIngress.Listening mismatch",
-			"ApplicationIngress", publishingStrategyIngress,
-			"IngressController", ingressController)
+/* Compares the static spec fields of the desired Spec against the existing IngressController's spec.
+Both the Domain and EndpointPublishingStrategy fields in the IngressController spec are static. This means
+that editing them will have no effect. Instead, the CR must be fully deleted and recreated with the desired
+Domain and EndpointPublishingStrategy filled in. Returns false if at least one of the existing fields don't match the desired
+*/
+
+func validateStaticSpec(ingressController operatorv1.IngressController, desiredSpec operatorv1.IngressControllerSpec) bool {
+	if !(desiredSpec.Domain == ingressController.Spec.Domain) {
 		return false
 	}
-	if ingressController.Status.EndpointPublishingStrategy.LoadBalancer == nil && capListening == "External" {
-		return true
-	}
-	if capListening != string(ingressController.Status.EndpointPublishingStrategy.LoadBalancer.Scope) {
-		log.Info("ApplicationIngress.Listening mismatch",
-			"ApplicationIngress", publishingStrategyIngress,
-			"IngressController", ingressController)
+	if !(desiredSpec.EndpointPublishingStrategy.LoadBalancer.Scope == ingressController.Spec.EndpointPublishingStrategy.LoadBalancer.Scope) {
 		return false
 	}
-	if publishingStrategyIngress.RouteSelector.MatchLabels == nil {
-		return true
+
+	return true
+}
+
+/* Compares the patchable desired Spec against the existing IngressController's status.
+The only patchable field that the ApplicationIngress controlls thats also stored in the IngressController's
+status is the RouteSelector. The default IngressController does not have those spec fields
+filled in at all when fisrt created. Instead, the default CRs status holds the correct values. This function
+is meant to be used when the default CRs spec is not filled in to see if the desired configuration is present in
+its status instead. Returns false if the RouteSelector in the status is empty or does not match the desired Spec.
```

Comment:
When does it return true? When the route selector is set or matches the desired spec? What is second field in the return value?

Reply:
The function returns a boolean, and is called validate. True returns when the status is valid, ie matches the desired spec.